Factories/src/Factory.cc

```diff
@@ -0,0 +1,382 @@
+    std::cout << "Extracting list of branches from skeleton..." << std::endl;
+    std::unique_ptr<TChain> t(new TChain("t"));
+    t->Add(skeleton.c_str());
+
+    // Get list of all branches
+    TObjArray* root_tree_branches = t->GetListOfBranches();
+    for (size_t i = 0; i < static_cast<size_t>(root_tree_branches->GetEntries()); i++) {
+        TBranch* b = static_cast<TBranch*>(root_tree_branches->UncheckedAt(i));
+
+        Branch branch;
+        branch.name = b->GetName();
+        branch.type = b->GetClassName();
+
+        if (branch.type.empty()) {
+            TLeaf* leaf = b->GetLeaf(branch.name.c_str());
+            if (! leaf) {
+                std::cerr << "Error: can't deduce type for branch '" << branch.name << "'" << std::endl;
+                continue;
+            }
+            branch.type = leaf->GetTypeName();
+        }
+
+        m_tree_branches.emplace(branch.name, branch);
+    }
+    std::cout << "Done." << std::endl;
+
+    std::FILE* f = std::fopen(config_file.c_str(), "r");
+    if (!f) {
+        throw std::runtime_error("Failed to open '" + config_file + "'");
+    }
+
+    Py_Initialize();
+
+    // Get a reference to the main module
+    // and global dictionary
+    PyObject* main_module = PyImport_AddModule("__main__");
+    m_global_dict = PyModule_GetDict(main_module);
+
+    // If PyROOT is used inside the script, it performs some cleanups when the python env. is destroyed. This cleanup makes ROOT unusable afterwards.
+    // The cleanup function is registered with the `atexit` module.
+    // The solution is to not execute the cleanup function. For that, before destroying the python env, we check the list of exit functions,
+    // and delete the one from PyROOT if found
+
+    // Ensure the module is loaded
+    PyObject* atexit_module = PyImport_ImportModule("atexit");
+
+    // Execute the script
+    PyObject* script_result = PyRun_File(f, config_file.c_str(), Py_file_input, m_global_dict, m_global_dict);
+
+    if (! script_result) {
+        PyErr_Print();
+        throw std::runtime_error("");
+    }
+
+    // Parse include path, include files
+    PyObject* py_include_dirs = PyDict_GetItemString(m_global_dict, "include_directories");
+    if (py_include_dirs) {
+        // Supposed to be a list of string, where each entry is a new include directory
+
+        if (! PyList_Check(py_include_dirs)) {
+            throw std::runtime_error("The 'include_directories' variable is not a list");
+        }
+
+        size_t l = PyList_Size(py_include_dirs);
+
+        for (size_t i = 0; i < l; i++) {
+            PyObject* item = PyList_GetItem(py_include_dirs, i);
+            if (! PyString_Check(item) ) {
+                throw std::runtime_error("The items of the 'include_directories' list must be strings");
+            }
+
+            std::string directory = PyString_AsString(item);
+            if (!fs::exists(directory)) {
```

Comment:
My idea here is that it's way easier to convert relative path to absolute path when you know where the relative path starts, and it's can be done super easy on the python side, don't you think?